### PR TITLE
feat(osc): odc user can assign lock db user when create osc task

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-feature
module-online schema change
#### What this PR does / why we need it:
assign db user(s) to lock exactly between swap table name in osc task, and this does not affects others db users

#### Which issue(s) this PR fixes:

1. add api `listUsers` to query users in selected datasource
2. modify `/database/databases/{id}` add column `lockDatabaseUserRequired` which indicate current database should lock db user between swap table
3. modify `createFlowInstance` add column `lockUsers` which indicate users to locked
4. modify `LockUserInterceptor`, old way is lock all users except db administrator,  new way is lock db user from odc user input

this feature has coordinate with font end.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java [hunk 1]
```diff
@@ -574,6 +605,25 @@ public Set<String> filterUnAuthorizedDatabaseNames(Set<String> databaseNames, @N
                 .collect(Collectors.toSet());
     }
 
+    @SkipAuthorize("internal authorized")
+    public Page<DataBaseUser> listUsers(Long dataSourceId) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java:609]: if listAll, List may better than Page

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java [hunk 2]
```diff
@@ -631,4 +681,23 @@ private String getLockKey(@NonNull Long connectionId) {
         return "DataSource_" + connectionId;
     }
 
+    private Pair<ConnectionSession, Boolean> getConnectionSessionLockUserRequiredPair(
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java:684]: too complex, avoid return Pair if not neccessary, how about return boolean instead ?

File: server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java [hunk 3]
```diff
@@ -164,6 +164,7 @@ public enum ErrorCodes implements ErrorCode {
     OscSqlTypeInconsistent,
     NoUniqueKeyExists,
     OscNotEnabled,
+    OscNotSelectLockUser,
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java:167]: Okey, font-end has restriction input parameters, back-end add restriction for enhancement

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java (hunk 2 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java:684]: okey,  ConnectionSession can be close  after predicate lock user or not

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java:609]: this method definition stay the same style like `/api/v2/database/databases` or other previous api

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**